server/backend/db_backend.go

```diff
@@ -25,11 +25,12 @@ import (
 var wei = big.NewInt(1000000000000000000)
 
 type MongoBackend struct {
-	host         string
-	mongo        *mgo.Database
-	mongoSession *mgo.Session
-	goClient     *goclient.Client
-	Lgr          *zap.Logger
+	host            string
+	mongo           *mgo.Database
+	mongoSession    *mgo.Session
+	goClient        *goclient.Client
+	Lgr             *zap.Logger
+	databaseVersion int
```

Comment:
Accessing this field is racey. It needs to be protected by a `sync.RWMutex`, or accessed via `sync/atomic` funcs (which may also require special placement within the struct fields?).

Reply:
yep, added mutex and removed updating this field in migrator (unnecessary since it's in grabber)